flycheck.el

```diff
@@ -6924,7 +6924,16 @@ See URL `http://www.erlang.org/'."
 
 (defun flycheck-rebar3-project-root (&optional _checker)
   "Return directory where =rebar.config= is located."
-  (locate-dominating-file buffer-file-name "rebar.config"))
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
```

Comment:
@joedevivo I see, but I'm not convinced that a regex is the right tool to use here, and I do not like string substitutions over pathnames.  That's just inherently fragile—for instance your current regex obviously breaks on windows pathnames.

How about the following algorithm instead:

1. Find the nearest `project-directory` containing `rebar.config` file with `locate-dominating-file`
2. From that file, find the nearest directory that contains a `_build` directory with `locate-dominating-file`.  If none exists return `project-directory` as the final working directory.
3. If one exists `let`-bind it to `default-directory` and then recursively call `flycheck-rebar3-project-root` again to restart the search for the top-most rebar project directory.

What do you think?